### PR TITLE
Allow additional function arguments in transform

### PR DESCRIPTION
**Pull request by robert8138**
When using `expression.transform`, we often pass in a function. It would be more flexible if we can allow users to pass in additional arguments to the function to override certain behaviors.

@tobymao, PTAL

### CODE REVIEW

File: sqlglot/expressions.py [hunk 1]
```diff
@@ -193,7 +193,7 @@ def to_s(self, hide_missing=True, level=0):
 
         return indent + left + right
 
-    def transform(self, fun, copy=True):
+    def transform(self, fun, copy=True, **fun_args):
```
Comment (tobymao) [inline sqlglot/expressions.py:196]: make it *args, **kwargs

Comment (robert8138) [reply sqlglot/expressions.py:196]: Good call.

Comment (robert8138) [reply sqlglot/expressions.py]: Updated.

File: benchmarks/bench.py [hunk 2]
```diff
@@ -48,10 +48,10 @@
 short = "select 1 as a, case when 1 then 1 when 2 then 2 else 3 end as b, c from x"
 
 crazy = "SELECT 1+"
-crazy += '+'.join(str(i) for i in range(500))
-crazy += ' AS a, 2*'
-crazy += '*'.join(str(i) for i in range(500))
-crazy += ' AS b FROM x'
+crazy += "+".join(str(i) for i in range(500))
```
Comment (robert8138) [inline benchmarks/bench.py:51]: These are just formatting changes from black

**Review by tobymao — APPROVED** [review]

**Pull request merged**